From bceb14c3ebdf6db13414e90390401786309b772c Mon Sep 17 00:00:00 2001
From: Eduardo Mendi <eduardox.mendi@intel.com>
Date: Thu, 2 May 2013 11:58:40 +0300
Subject: [PORT FROM MAIN] Telephony: In Double call, ringtone must keep in earpiece

BZ: 80477

The ringtone can be heard in IHF when receive and answer a second voice call.
The root cause is that the function setAudioMode is called upon accepting
a second call while a first call is active. In this case, the audio mode is
MODE_IN_CALL and should remain MODE_IN_CALL. However, in function answerCall
the setAudioMode is called before that the phone state of the waitting phone
is changed from WAITING to ACTIVE. Then the getState function state returns
state RINGING which leads to wrong audio mode configuration.

Change-Id: I400935a69abb2458d5c96ab5587efea06ad43825
Orig-Change-Id: I4b776d59d0a923eeb9d98a9e0bb3d6d2626889ec
Category: device enablement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: no, proprietary
Signed-off-by: Eduardo Mendi <eduardox.mendi@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../android/internal/telephony/CallManager.java    |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/java/com/android/internal/telephony/CallManager.java b/src/java/com/android/internal/telephony/CallManager.java
index 69a7b90..2e446e6 100644
--- a/src/java/com/android/internal/telephony/CallManager.java
+++ b/src/java/com/android/internal/telephony/CallManager.java
@@ -381,7 +381,7 @@ public final class CallManager {
         switch (getState()) {
             case RINGING:
                 int curAudioMode = audioManager.getMode();
-                if (curAudioMode != AudioManager.MODE_RINGTONE) {
+                if ((!hasActiveFgCall()) && (curAudioMode != AudioManager.MODE_RINGTONE)) {
                     // only request audio focus if the ringtone is going to be heard
                     if (audioManager.getStreamVolume(AudioManager.STREAM_RING) > 0) {
                         if (VDBG) Rlog.d(LOG_TAG, "requestAudioFocus on STREAM_RING");
-- 
1.7.4.1

